### DISCUSSION TITLE
Styling tab-{bar,line}-mode

### DISCUSSION THREAD

**Discussion by gramanas** [tarsius/moody#44] [Ideas]
Hello,

Are there any plans for adding support for styling the new tab modes with the same moody style?

Thanks!

**Comment by tarsius** [comment]
Yes I plan to do that eventually but probably not until I have at least experimented with using `tab-bar-mode` or `tab-line-mode` myself.

**Comment by haji-ali** [comment]
I've managed to use moody for `tab-line` like this:

```

(setq tab-line-tab-name-function  #'my/tab-line-tab-name-buffer)

(defun my/tab-line-tab-name-buffer (buffer &optional buffers)
  (apply
   'concat
   (moody-tab
    (tab-line-tab-name-buffer buffer buffers) nil 'up)))
```

The results are passable. The only issue, which is quite fundamental, is that a mouse highlight face which changes the background of the tabs looks weird because moody tabs do not change colour (being xpm images) while the text in between them changes color.

**Comment by tarsius** [comment]
For the last month I have been trying to take a break and failed. This is now the last discussion, issue or pull-request that remains open in any of my repositories beside the big three (magit, transient and forge). To reduce the pressure to get this done now, despite not feeling like doing so and really needing a break, I am closing this. I still don't use tab-...-mode. I still intend to do so eventually. And once I do, I am certain to want to moodify it. So its still likely that this will happen eventually, and I do not need to leave this open as a reminder.
